tests/component/records/schema/field/test_pandas.py

```diff
@@ -0,0 +1,110 @@
+from records_mover.records.schema.field.constraints import (
+    RecordsSchemaFieldConstraints,
+    RecordsSchemaFieldStringConstraints,
+    RecordsSchemaFieldIntegerConstraints,
+    RecordsSchemaFieldDecimalConstraints,
+)
+from records_mover.records.schema.field.statistics import (
+    RecordsSchemaFieldStringStatistics,
+)
+from records_mover.records.schema.field.representation import RecordsSchemaPandasFieldRepresentation
+from records_mover.records.schema.field import RecordsSchemaField
+from records_mover.records.schema.field.pandas import refine_field_from_series
+from records_mover.records.schema.field.types import RECORDS_FIELD_TYPES
+import pandas as pd
+
+
+class TestPandas(unittest.TestCase):
+    def test_refine_field_from_series_more_specific(self) -> None:
+        # This test is designed to break when a new field type is
+        # introduced, so you can add new expectations and make sure
+        # the code handles the new type!
+
+        fields = {
+            'integer': {
+                'series': pd.Series([30, 35, 40]),
+                'constraints_type': RecordsSchemaFieldIntegerConstraints,
+                'statistics_type': type(None),
+            },
+            'decimal': {
+                'series': pd.Series([30.0, 35.1, 40.2]),
+                'constraints_type': RecordsSchemaFieldDecimalConstraints,
+                'statistics_type': type(None),
+            },
+            'string': {
+                'series': pd.Series(['a', 'b', 'c']),
+                'constraints_type': RecordsSchemaFieldStringConstraints,
+                'statistics_type': RecordsSchemaFieldStringStatistics,
+            },
+            'boolean': {
+                'series': pd.Series([True, True, False]),
+                'constraints_type': RecordsSchemaFieldConstraints,
+                'statistics_type': type(None),
+            },
+            'date': {
+                'series': pd.Series([datetime.date(2020, 1, 1)]),
+                'constraints_type': RecordsSchemaFieldConstraints,
+                'statistics_type': type(None),
+            },
+            'time': {
+                'series': pd.Series([datetime.time(hour=12, minute=0, second=0)]),
+                'constraints_type': RecordsSchemaFieldConstraints,
+                'statistics_type': type(None),
+            },
+            'timetz': {
+                'series': pd.Series([datetime.time(hour=12, minute=0, second=0,
+                                                   tzinfo=pytz.timezone('US/Eastern'))]),
+                # TODO: File issue for backlog about not being able to
+                # derive a time with timezone
+                'expected_field_type': 'time',
+                'constraints_type': RecordsSchemaFieldConstraints,
+                'statistics_type': type(None),
+            },
+            'datetime': {
+                'series': pd.Series([datetime.datetime(2020, 1, 1, hour=12)]),
+                # TODO: Figure out why this is
```

Comment:
Uncompleted punchlist item detected--consider resolving or moving this to your issue tracker